content/en/docs/Reference/file-based-catalogs.md

```diff
@@ -0,0 +1,586 @@
+(named <dcDirName>.Dockerfile) that can be used to build the index. If a
+Dockerfile with the same name already exists, this command will fail.
+
+When specifying extra labels, note that if duplicate keys exist, only the last
+value of each duplicate key will be added to the generated Dockerfile.
+
+Usage:
+  opm alpha generate dockerfile <dcRootDir> [flags]
+
+Flags:
+  -i, --binary-image string    Image in which to build catalog. (default "quay.io/operator-framework/upstream-opm-builder")
+  -l, --extra-labels strings   Extra labels to include in the generated Dockerfile. Labels should be of the form 'key=value'.
+  -h, --help                   help for dockerfile
+
+Global Flags:
+      --skip-tls   skip TLS certificate verification for container image registries while pulling bundles or index
+```
+
+## Guidelines
+
+### Immutable bundles
+
+OLM's general advice is that bundle images and their metadata should be treated as immutable. If a broken bundle has
+been pushed to an index, you must assume that at least one of your users has upgraded to that bundle. Based on that
+assumption, you must release another bundle with an upgrade edge from the broken bundle to ensure users with the broken
+bundle installed receive an upgrade. OLM will not reinstall an installed bundle if the contents of that bundle are
+updated in the index.
+
+However, there are some cases where a change in the index metadata is preferred. For example:
+- Channel promotion - if you already released a bundle and later decide that you'd like to add it to another channel,
+  simply add an entry for your bundle in another `olm.channel` blob.
+- New upgrade edges - if you release a new 1.2.z (e.g. 1.2.4), but 1.3.0 is already released, you can update the index
+  metadata for 1.3.0 to skip 1.2.4.
+
+### Use of source control
+
+OLM highly recommends storing index metadata in source control and treating the source-controlled metadata as the source
+of truth. Updates to index images should:
+1. Update the source-controlled index directory with a new commit.
+2. Build and push the index image. OLM suggests using a consistent tagging taxonomy (e.g. `:latest` or
+   `:<targetClusterVersion>` so that users can receive updates to an index as they become available.
+
+<!--
+TODO(joelanford): Add a link to an file-based catalog repository when one exists in the future.
+-->
+
+## Examples
+### Building a composite catalog
+
+With file-based catalogs, catalog maintainers can focus on operator curation and compatibility.
+Since operator authors have already produced operator-specific indexes for their operators, catalog
+maintainers can build their catalog simply by rendering each operator index into a subdirectory of the
+catalog's root index directory.
+
+There are many possible ways to build a catalog, but an extremely simple approach would be to:
+
+1. Maintain a single configuration file containing image references for each operator in the catalog
```

Comment:
This also renders somewhat strangely like in the previous comment - I'm not sure what's the best way to structure this though besides making them list items using the `-` syntax.